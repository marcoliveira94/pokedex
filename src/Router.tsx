import React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { RoutesEnum } from './models/routes.enum'

// Routes Import
import DetailsScreen from './screens/Details/Details.screen'
import HomeScreen from './screens/Home/Home.screen'
import { Pokemon } from './models/pokemonList.type'

const Stack = createStackNavigator()

export type RootStackParamList = {
  Fact: undefined
  Details: Pokemon | undefined
  Home: undefined
}

const Router = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name={RoutesEnum.HOME} component={HomeScreen} />
        <Stack.Screen name={RoutesEnum.DETAILS} component={DetailsScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  )
}

export default Router
