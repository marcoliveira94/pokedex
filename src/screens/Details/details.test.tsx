import React from 'react'
import DetailsScreen from './Details.screen'

import renderer from 'react-test-renderer'
import { StackNavigationProp } from '@react-navigation/stack'
import { RootStackParamList } from '../../Router'
import { RouteProp } from '@react-navigation/native'

test('renders correctly', () => {
  const navigation = ({
    navigate: jest.fn()
  } as unknown) as StackNavigationProp<RootStackParamList, 'Details'>

  const route = ({ navigate: jest.fn() } as unknown) as RouteProp<
    RootStackParamList,
    'Details'
  >

  const tree = renderer
    .create(<DetailsScreen navigation={navigation} route={route} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
