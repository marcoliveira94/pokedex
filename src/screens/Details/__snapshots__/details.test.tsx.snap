// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="sc-AxjAm bvfVxZ"
>
  <ActivityIndicator
    animating={true}
    color="red"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
